### PR TITLE
🔥 remove redundant backend logic

### PR DESCRIPTION
**Pull request by dorlugasigal**
after a quick chat with @Mimetis, decided to remove `aad_client` and its usages and dependencies since its not used.
the library will migrate into backend token verification only

- [x] removed UI code
- [x] removed aad_client code
- [x] fixed tests

### CODE REVIEW

File: tests/test_aad_helpers.py [hunk 1]
```diff
@@ -4,8 +4,8 @@
 from authlib.jose.rfc7519.claims import JWTClaims
 from helpers import gen_payload
 
-from aad import AadUser, AuthError, AzureAdSettings
-from aad.aad_helpers import _decode_token, _get_user_from_claims, _validate_claims
+from src.aad_fastapi import AadUser, AuthError, AzureAdSettings
+from src.aad_fastapi.aad_helpers import _decode_token, _get_user_from_claims, _validate_claims
```
Comment (SharonHart) [inline tests/test_aad_helpers.py]: managed to run in w/o src. just aad_fastapi.aad_helpers

Comment (dorlugasigal) [reply tests/test_aad_helpers.py]: fixed for all tests files (had to set up source root in IDE)

**Comment by Mimetis** [comment]
Hey @dorlugasigal 
Thanks for working on this huge part!

I'm currently looking at what you've removed.
As we want to remove everything that is not related to backend auth, I agree to remove the main.py sample, but now we don't have any sample with the **e2e** scenario.
Do you think we can add a new main.py file using a classic `Flask` / `MSAL` web application?

**Comment by dorlugasigal** [comment]
> we do have a sample, the API sample.
I only removed the UI sample which isn't relevant anymore, all the user needs to do is prepare a local.env file with the appropriate values for their app registrations

**Comment by dorlugasigal** [comment]
> we do have a sample, the API sample.
> I only removed the UI sample which isn't relevant anymore, all the user needs to do is prepare a local.env file with the appropriate values for their app registrations



> Hey @dorlugasigal Thanks for working on this huge part!
> 
> I'm currently looking at what you've removed. As we want to remove everything that is not related to backend auth, I agree to remove the main.py sample, but now we don't have any sample with the **e2e** scenario. Do you think we can add a new main.py file using a classic `Flask` / `MSAL` web application?

@Mimetis 
we do have a sample, the API sample.
I only removed the UI sample which isn't relevant anymore, all the user needs to do is prepare a local.env file with the appropriate values for their app registrations

**Comment by Mimetis** [comment]
Ok, that means, we are relying on the Swagger page (`http://localhost:8000/docs`) to test the secured endpoints we are providing as sample, right?

**Comment by dorlugasigal** [comment]
> Ok, that means, we are relying on the Swagger page (`http://localhost:8000/docs`) to test the secured endpoints we are providing as sample, right?

@Mimetis Correct (as commonly used by backend developers)

**Comment by dorlugasigal** [comment]
@Mimetis I've updated the sample `README.md` to explain how to run it, and removed any reference for the UI part

File: sample/api/main.py [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
+
 from dotenv import load_dotenv
 from fastapi import Depends, FastAPI
-from routers import engines
 from starlette.middleware.authentication import AuthenticationMiddleware
 from starlette.requests import Request
 
-from aad_fastapi import (
+from sample.api.routers import engines
+from src.aad_fastapi import (
```
Comment (SharonHart) [inline sample/api/main.py]:
```suggestion
from aad_fastapi import (
```

File: tests/test_oauth2_scheme.py [hunk 3]
```diff
@@ -39,11 +39,12 @@ def test_oauth2_scheme_custom_options_is_set():
         == "https://login.microsoftonline.com/tenant_id/oauth2/v2.0/authorize"
     )
     assert len(oauth2.model.flows.authorizationCode.scopes) == 2
+    print(oauth2.model.flows.authorizationCode.scopes)
```
Comment (SharonHart) [inline tests/test_oauth2_scheme.py]:
```suggestion
```

File: tests/requirements.txt [hunk 4]
```diff
@@ -1 +1,3 @@
 async-asgi-testclient==1.4.6
+pytest
+pytest-asyncio
```
Comment (SharonHart) [inline tests/requirements.txt:3]: 👍

**Review by SharonHart — APPROVED** [review]
Left small comments

**Pull request merged**